presto-delta/src/main/java/com/facebook/presto/delta/DeltaSessionProperties.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.delta;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static com.facebook.presto.hive.HiveSessionProperties.dataSizeSessionProperty;
+import static com.facebook.presto.spi.session.PropertyMetadata.booleanProperty;
+
+public final class DeltaSessionProperties
+{
+    public static final String PARTITION_PRUNING_ENABLED = "partition_pruning_enabled";
+    public static final String FILTER_PUSHDOWN_ENABLED = "filter_pushdown_enabled";
+    private static final String CACHE_ENABLED = "cache_enabled";
```

Comment:
Do we need `cache_enabled` session property since it's value is not getting used as of now?

Reply:
@vkorukanti what tests use it? It's better to add properties back when caching is actually supported. Can you modify the tests not to use it?

Reply:
Without this property the integration tests fail with the following error callstack. The problem is [here](https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/cache/HiveCachingHdfsConfiguration.java#L76), we always look for the property `CACHE_ENABLED`. If not present it throws an unknown session property error. This is not related to Delta. In any case, caching will be added once this PR lands in master.


```com.facebook.presto.spi.PrestoException: Unknown session property delta.cache_enabled

	at com.facebook.presto.metadata.SessionPropertyManager.lambda$decodeCatalogPropertyValue$1(SessionPropertyManager.java:181)
	at java.util.Optional.orElseThrow(Optional.java:290)
	at com.facebook.presto.metadata.SessionPropertyManager.decodeCatalogPropertyValue(SessionPropertyManager.java:181)
	at com.facebook.presto.FullConnectorSession.getProperty(FullConnectorSession.java:150)
	at com.facebook.presto.hive.HiveSessionProperties.isCacheEnabled(HiveSessionProperties.java:1087)
	at java.util.Optional.map(Optional.java:215)
	at com.facebook.presto.hive.cache.HiveCachingHdfsConfiguration.lambda$getConfiguration$0(HiveCachingHdfsConfiguration.java:76)
	at com.facebook.presto.hive.cache.HiveCachingHdfsConfiguration$CachingJobConf.createFileSystem(HiveCachingHdfsConfiguration.java:105)
	at org.apache.hadoop.fs.PrestoFileSystemCache.get(PrestoFileSystemCache.java:59)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:373)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
	at com.facebook.presto.hive.HdfsEnvironment.lambda$getFileSystem$0(HdfsEnvironment.java:71)
	at com.facebook.presto.hive.authentication.NoHdfsAuthentication.doAs(NoHdfsAuthentication.java:23)
	at com.facebook.presto.hive.HdfsEnvironment.getFileSystem(HdfsEnvironment.java:70)
	at com.facebook.presto.hive.HdfsEnvironment.getFileSystem(HdfsEnvironment.java:64)
	at com.facebook.presto.delta.DeltaClient.loadDeltaTableLog(DeltaClient.java:148)
	at com.facebook.presto.delta.DeltaClient.getTable(DeltaClient.java:79)
	at com.facebook.presto.delta.DeltaMetadata.getTableHandle(DeltaMetadata.java:136)
	at com.facebook.presto.delta.DeltaMetadata.getTableHandle(DeltaMetadata.java:57)
	at com.facebook.presto.metadata.MetadataManager.getTableHandle(MetadataManager.java:330)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer$Visitor.lambda$visitTable$17(StatementAnalyzer.java:1244)
`